### ISSUE TITLE
Order of pinned columns is wrong on restore of grid state

### ISSUE THREAD

**Issue by amit-kohli** [IgniteUI/igniteui-angular#6743]
## Description  
Pin a few columns in the grid. Please ensure that the order is different from the order in which they are displayed in the grid. For instance - if the grid columns are 1, 2, 3, 4, 5, 6, 7
then pin 5,3,1

 * igniteui-angular version: 8.2.18
 * browser: [‎2/‎14/‎2020 11:10 AM] Hutchinson, Brook /C: 
Google Chrome Version 80.0.3987.100 (Official Build) (64-bit) 

## Steps to reproduce  

Create a grid with column headers 1, 2, 3, 4, 5, 6, 7
then pin 5,3,1
Save the grid state to local storage (https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/grid.html#state-persistence)
Navigate away from the page and return to it.

## Result  
What is the actual result after following the steps to reproduce?
the columns are pinned but in the wrong order - the order they are displayed in is 1, 3, 5 but it 

## Expected result  
What is the expected result after following the steps to reproduce?
The order of the pinned columns should be 5, 3, 1

## Attachments  
Attach a sample if available, and screenshots, if applicable.

**Comment by hanastasov** [comment]
Hi @amit-kohli, I am not able to replicate the described behavior. I tested with https://stackblitz.com/edit/angular-9wg2cp following the steps provided and columns are restored in the expected 5,3,1 order. Please confirm if you managedto replicate the issue using this sample.

**Comment by amit-kohli** [comment]
Hi - Thanks for the sample - that works. The one difference I've noticed is that in the stored state between the sample and my code - the order of columns is different. In your sample the pinned columns are first whereas in my code they are persisted in the order they are set up in the template with pinned=true. 
I had adapted the code from the igxGridState directive so can just hack it further when getting the columns to first get the ones that have pinned === true.
Thank you,
Amit
